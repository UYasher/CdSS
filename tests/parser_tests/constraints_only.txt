c(x == y)
c(2 * x + x - x == y)
c(x // 2 == y)
c(x >= y)
c(x <= y)
c(x < y)
c(x > y)
c(x != y)
c(x == -y)
c(x == --y)