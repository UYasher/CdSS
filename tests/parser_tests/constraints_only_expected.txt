start
  instruction
    constraint
      eq
        unit
          punit
            var
              string    x
        unit
          punit
            var
              string    y
  instruction
    constraint
      eq
        sub
          add
            unit
              mul
                punit
                  number        2
                var
                  string        x
            punit
              var
                string  x
          punit
            var
              string    x
        unit
          punit
            var
              string    y
  instruction
    constraint
      eq
        unit
          div
            punit
              var
                string  x
            number      2
        unit
          punit
            var
              string    y
  instruction
    constraint
      geq
        unit
          punit
            var
              string    x
        unit
          punit
            var
              string    y
  instruction
    constraint
      leq
        unit
          punit
            var
              string    x
        unit
          punit
            var
              string    y
  instruction
    constraint
      lt
        unit
          punit
            var
              string    x
        unit
          punit
            var
              string    y
  instruction
    constraint
      gt
        unit
          punit
            var
              string    x
        unit
          punit
            var
              string    y
  instruction
    constraint
      neq
        unit
          punit
            var
              string    x
        unit
          punit
            var
              string    y
  instruction
    constraint
      eq
        unit
          punit
            var
              string    x
        unit
          punit
            neg
              var
                string  y
  instruction
    constraint
      eq
        unit
          punit
            var
              string    x
        unit
          punit
            neg
              neg
                var
                  string        y